.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,22 @@
+name: Continuous Integration
+on:
+  push:
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    name: Build and Run Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 #v3.5.3
```

Comment:
I'm personally a fan of more liberal versioning - e.g. `@v3`; you are more at risk of changings breaking something, but equally you will pick up patches automatically (and it's much easier to see if you are out of date)

Reply:
I prefer to use a sha in accordance with the GitHub security guidelines https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

It's something I made a script to enforce at work because it's too easy for a tag to be moved and completely derail your pipeline.

With that said, as this is a GitHub action it's probably not required, but I'm in the habit now 🤣